### PR TITLE
Feature/is equal

### PR DESCRIPTION
**Pull request by dictor93**
### Description:
Created isEqual function which deeply compares two items

### New features:
- [x] isEqual

### PR status:
- [x] Version bumped
- [x] Change-log updated
- [x] Tests added or updated
- [x] PR keeps 100% test coverage
- [x] Type definition updated
- [x] Readme updated
- [x] Linter is applied
- [x] Const arrow functions are used instead of pure function definitions where applicable
- [x] Functions are listed in alphabetic order in index.js, index.d.ts and readme

### CODE REVIEW

**Review by oliverbitfinex — APPROVED** [review]

**Review by ezewer — APPROVED** [review]
looks good to me!

File: src/isEqual.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+'use-strict'
+
+const { objectsDeepEqual, strictEqual } = require('./util')
```
Comment (ezewer) [inline src/isEqual.js]: Add directly using: 
const objectsDeepEqual = require('./objectsDeepEqual')
const strictEqual = require('./strictEqual')

File: src/util/index.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+'use strict'
```
Comment (ezewer) [inline src/util/index.js]: Delete this file!

**Review by ezewer — CHANGES_REQUESTED** [review]
Required changes

**Review by ezewer — APPROVED** [review]
looks good to me!

**Review by ezewer — APPROVED** [review]
looks good to me!

File: src/util/objectsDeepEqual.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
+'use-strict'
```
Comment (vigan-abd) [inline src/util/objectsDeepEqual.js:1]: this file doesn't respect rule ` PR keeps 100% test coverage`, @dictor93 please open a PR with remaining cases

Comment (vigan-abd) [reply src/util/objectsDeepEqual.js:1]: same goes for isEqual file

**Pull request merged**